cmd/main.go

```diff
@@ -2,43 +2,52 @@ package main
 
 import (
 	"flag"
+	"net/http"
+	"os"
+	"regexp"
+
 	"github.com/ViaQ/logerr/log"
+	"github.com/fsnotify/fsnotify"
 	"github.com/log-file-metric-exporter/pkg/symnotify"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"net/http"
-	"os"
-	"regexp"
 )
 
 var (
-	verbosity int = 0
-	//Reference regexp https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/blob/master/lib/fluent/plugin/filter_kubernetes_metadata.rb#L56, https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/kubelet/dockertools/docker.go
-	//compile k8 logfilepathname pattern
-	kubernetesregexpCompiled = regexp.MustCompile(`.var.log.containers.([a-z0-9][-a-z0-9]*[a-z0-9])_([^_]+)_(.+)-([a-z0-9]{64})\.log$`)
+	verbosity int = 1
+	logDir        = "/var/log/pods"
+	logFile       = regexp.MustCompile(`/([a-z0-9-]+)_([a-z0-9-]+)_([a-f0-9-]+)/([a-z0-9-]+)/.*\.log`)
 )
 
 const (
-	PodNameIndex = iota + 1
-	NamespaceIndex
+	NamespaceIndex = iota + 1
+	PodNameIndex
+	PodUUIDIndex
 	ContainerNameIndex
-	DockerIndex
-	MatchLen
 )
 
 type FileWatcher struct {
 	watcher *symnotify.Watcher
 	metrics *prometheus.CounterVec
 	sizes   map[string]float64
-	added   map[string]bool
 }
 
-func (w *FileWatcher) Update(path string, namespace string, podname string, containername string) error {
-	var add float64
-	var lastSize float64
-	var size float64
+func (w *FileWatcher) Update(path string) (err error) {
+	defer func() {
+		if os.IsNotExist(err) {
+			w.Forget(path)
+			err = nil // Not an error if a file disappears
+		}
+		if err != nil {
+			log.Error(err, "error updating metric", "path", path)
+		}
+	}()
 
-	counter, err := w.metrics.GetMetricWithLabelValues(path, namespace, podname, containername)
+	match := logFile.FindStringSubmatch(path)
+	if match == nil {
+		return nil // Ignore non-log files
+	}
+	counter, err := w.metrics.GetMetricWithLabelValues(path, match[NamespaceIndex], match[PodNameIndex], match[ContainerNameIndex])
```

Comment:
This is fine but contingent upon the regex which is specific to the directory.  We should either make everything configurable or none of it at all and make an assumption

Reply:
The regex no longer includes the initial directory part, that makes it easier to test since you can direct it to places other thna /var/log/pods

Reply:
The only thing that is currently configurable is the root of the log directory tree, the rest is hard coded. If we want to make it more configurable in future I think we should open a separate bug post release.